integrations/activewindow.cpp

```diff
@@ -0,0 +1,250 @@
+#include "core.h"
+#include <QCoreApplication>
+#include <QDBusConnection>
+#include <QDBusInterface>
+#include <QDBusReply>
+#include <QTemporaryFile>
+#include <QFile>
+#include <QTextStream>
+#include <QTimer>
+#include <QDebug>
+#include <QStandardPaths>
+#include <QDir>
+
+class ActiveWindowWatcher : public QObject
+{
+    Q_OBJECT
+    Q_CLASSINFO("D-Bus Interface", "org.davidedmundson.kiot.ActiveWindow")
+    
+public:
+    explicit ActiveWindowWatcher(QObject *parent = nullptr);
+
+public slots:
+    Q_SCRIPTABLE void UpdateTitle(const QString &title);
+
+private slots:
+    void onActiveWindowChanged(const QString &title);
+
+
+private:
+    bool registerKWinScript();
+    void publish(const QString &state);
+
+    Sensor *m_sensor;
+    QDBusInterface *m_kwinIface = nullptr;
+    QString m_lastTitle;
+    QString m_scriptPath;
+    bool m_connected = false;
+};
+
+ActiveWindowWatcher::ActiveWindowWatcher(QObject *parent)
+    : QObject(parent)
+{
+    m_sensor = new Sensor(this);
+    m_sensor->setId("active_window");
+    m_sensor->setName("Active Window");
+    m_sensor->setDiscoveryConfig("icon", "mdi:application");
+    m_sensor->setState("Loading...");
```

Comment:
I don't think this `setState("Loading...")` adds anything useful to the HA consumer side.